NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -768,60 +769,37 @@ NFIQ2UI::parseModel(const NFIQ2UI::Arguments &arguments)
 		modelInfoFilePath = arguments.flags.model;
 	}
 
-	// FIXME: Make below a separate function and make a Model class
-	static const std::string ModelInfoKeyName { "Name" };
-	static const std::string ModelInfoKeyTrainer { "Trainer" };
-	static const std::string ModelInfoKeyDescription { "Description" };
-	static const std::string ModelInfoKeyVersion { "Version" };
-	static const std::string ModelInfoKeyPath { "Path" };
-	static const std::string ModelInfoKeyHash { "Hash" };
-
-	std::unique_ptr<BE::IO::Properties> props;
-	std::string modelFile, hash;
-
-	try {
-		props.reset(new BE::IO::PropertiesFile(
-		    modelInfoFilePath, BE::IO::Mode::ReadOnly));
-	} catch (const BE::Error::Exception &e) {
-		throw NFIQ2UI::PropertyParseError(
-		    "Unable to parse model info file '" + modelInfoFilePath +
-		    "' (" + e.whatString() + ')');
-	}
-
+	std::shared_ptr<NFIQ::ModelInfo> modelInfoObj {};
 	try {
-		modelFile = props->getProperty(ModelInfoKeyPath);
-		if (modelFile.empty())
-			throw BE::Error::StrategyError("No path provided");
-	} catch (const BE::Error::Exception &e) {
-		throw NFIQ2UI::PropertyParseError("Unable to parse '" +
-		    ModelInfoKeyPath + "' from '" + modelInfoFilePath + "' (" +
-		    e.whatString() + ')');
+		modelInfoObj = std::make_shared<NFIQ::ModelInfo>(
+		    modelInfoFilePath);
+	} catch (const NFIQ::NFIQException &e) {
+		throw NFIQ2UI::FileNotFoundError(
+		    "Could not construct ModelInfo Object from: " +
+		    modelInfoFilePath);
 	}
 
 	// Path to model might be relative to the model info file, not the cwd
-	if ((modelFile.front() != '/') &&
-	    ((modelFile.length() > 2) && (modelFile.substr(0, 2) != "\\\\")) &&
-	    ((modelFile.length() > 3) && (modelFile.substr(1, 2) != ":\\")) &&
-	    ((modelFile.length() > 3) && (modelFile.substr(1, 2) != ":/"))) {
-		modelFile = BE::Text::dirname(modelInfoFilePath) + '/' +
-		    modelFile;
+	if ((modelInfoObj->getModelPath().front() != '/') &&
+	    ((modelInfoObj->getModelPath().length() > 2) &&
+		(modelInfoObj->getModelPath().substr(0, 2) != "\\\\")) &&
+	    ((modelInfoObj->getModelPath().length() > 3) &&
+		(modelInfoObj->getModelPath().substr(1, 2) != ":\\")) &&
+	    ((modelInfoObj->getModelPath().length() > 3) &&
+		(modelInfoObj->getModelPath().substr(1, 2) != ":/"))) {
+		modelInfoObj->setModelPath(
+		    BE::Text::dirname(modelInfoFilePath) + '/' +
+		    modelInfoObj->getModelPath());
 	}
```

Comment:
This should be part of parsing `ModelInfo`, which means you cannot use `BE::Text::dirname()`